src/components/select-field/BaseSelectField.js

```diff
@@ -200,11 +205,11 @@ class BaseSelectField extends React.Component<Props, State> {
         }
     };
 
-    handleButtonClick = () => {
+    handleButtonClick = (event: SyntheticEvent<HTMLElement>) => {
         if (this.state.isOpen) {
```

Comment:
as a more terse alternative, `this[this.state.isOpen ? 'closeDropdown' : 'openDropdown'](event)`